oauth2client/client.py

```diff
@@ -1819,9 +1822,14 @@ def __init__(self, client_id, client_secret, scope,
         proper multi-login session, thereby simplifying the login flow.
       device_uri: string, URI for device authorization endpoint. For convenience
         defaults to Google's endpoints but any OAuth 2.0 provider can be used.
+      authorization_header: string, For use with OAuth 2.0 providers that
+        require a client to authenticate using a header value instead of passing
+        client_secret in the POST body.
       **kwargs: dict, The keyword arguments are all optional and required
                         parameters for the OAuth calls.
     """
+    if scope is None:
+      raise TypeError("The value of scope must not be None")
```

Comment:
Then make it a positional argument. Don't re-invent the wheel here.

Or is the issue that you can't change argument order for backwards compatibility?